core/src/doc/check.rs

```diff
@@ -311,6 +316,47 @@ impl Document {
 	/// permissions are `NONE`. This function does not
 	/// check any custom advanced table permissions,
 	/// which should be checked at a later stage.
```

Comment:
I think this comment may have been a copy of that from `check_permissions_quick`. It would be ideal to describe that the purpose of this function is to check the permissions for viewing the document resulting from an operation of any kind based on the `SELECT` permission. This is at least how I understand it.